qiskit/_quantumprogram.py

```diff
@@ -1339,7 +1394,13 @@ def _run_internal(self, qobj_list, wait=5, timeout=60, callback=None):
                                      callback=callback)
         job_processor.submit()
 
-    def execute(self, name_of_circuits, backend="local_qasm_simulator",
+    def _create_id(self, formatstring, existing):
+        for i in range(10000):
+            if formatstring % i not in existing:
+                return formatstring % i
+        raise QISKitError("Anonymous identifier was not possible to be created")
```

Comment:
Instead of the explicit loop, perhaps it makes sense to use a generator (or several)  or similar, maybe with the help of [itertools.count](https://docs.python.org/3/library/itertools.html#itertools.count)?

```
>>> some_counter = itertools.count()
>>> next(some_counter)
0
>>> next(some_counter)
1
>>> next(some_counter)
2
```

Reply:
`range()` does not requires imports, creates an object `range` and there is a limit to the looping.

Reply:
Importing from the standard lib is "free", so it should not really weight too much in the decision - and actually the limit to the looping is not the best of practices, as it introduces some "arbitrary magic number" :tm: . If the standard library provides a pythonic way of solving a problem (in this case, generating sequential numbers), why not take advantage of it?

Reply:
done in ﻿﻿﻿41a20cd2948464d52c14b630de1c2aa28467bca7